crates/forge_domain/src/orch.rs

```diff
@@ -252,16 +258,16 @@ impl<A: Services> Orchestrator<A> {
             self.dispatch_spawned(event).await?;
             Ok(ToolResult::from(tool_call.clone()).success("Event Dispatched Successfully"))
         } else {
-            Ok(self
-                .services
+            self.services
                 .tool_service()
                 .call(
                     ToolCallContext::default()
                         .sender(self.sender.clone())
                         .agent_id(agent.id.clone()),
                     tool_call.clone(),
+                    self.conversation.read().await.mcp.clone(),
```

Comment:
ToolCallContext was created for the purpose of passing relevant context to the caller. Lets select the actual stuff that's required, and copy it in ToolCallContext